content/introduction/how-ipfs-works.md

```diff
@@ -0,0 +1,88 @@
+
+This is where the [IPLD project](https://ipld.io/) comes in. **Hashes identify content, and IPLD links that content together** using any number of arrangements. It's a bit like your file system at home. You might organize all your bills by year; another person might organize them by type. These are both valid ways of organizing content, but you have to know which one you're using, otherwise you can't find anything or add anything to your folders.
+
+IPLD is "software" that provides both rules for how to organize content-addressed data, and translations across different organization methods. Since different distributed systems structure their data in different ways, IPLD provides a platform for combining pluggable modules (parsers for each possible type of IPLD node) to resolve a path, selector, or query across many linked nodes (of varying formats, even!). IPLD really refers to any content addressable data with links to other content addressable data.
+
+In the case of a peer to peer network using the IPFS protocol, your computer uses "IPFS-flavored IPLD" to get from raw content to an IPFS address. IPFS has its own preferences and conventions about how data should be broken up into a DAG (more on DAGs below!); IPLD links content on the IPFS network together using those conventions.
+
+**Everything else in the IPFS ecosystem builds on top of this core concept; linked, addressable content is the fundamental connecting element that makes the rest work.**
+
+### 2 \\ DAGs
+
+<img alt="IPFS Stack - Applications" src="../assets/ipfs_stack-apps.png" width="200px" />
+
+IPFS and many other distributed systems take advantage of a concept called [directed acyclic graphs](https://en.wikipedia.org/wiki/Directed_acyclic_graph), or DAGs. DAG structures are what allow distributed nodes to connect to each other in computationally reasonable ways, and are how you can build trees of content-addressed nodes.
+
+As mentioned above, IPFS has its own preferences and conventions about how data should be broken up into DAGs that optimize for its specific use cases. You can do this chunking and structuring in lots of different ways. For example, Git is a DAG that has many versions of your repo inside of it. Its DAG tree doesn't behave in the same way the IPFS DAG does.
+
+IPFS first separates your content into _blocks_. If you had a huge file with a single CID, you'd have to transfer the whole CID every time someone requested it. But Instead, you can break it into blocks, and transfer it block by block instead of one massive chunk at a time. Each block might come from a different source, even! This is a much more efficient way to transmit data. (If you've used BitTorrent, you may have noticed that when you download a file, it fetches it from a bunch of random people; this is the same idea.)
```

Comment:
```suggestion
IPFS first separates your content into _blocks_. If you had a huge file with a single CID, you'd have to transfer the whole file in a single chunk every time someone requested it. But Instead, you can break it into blocks, and transfer it block by block. Each block might come from a different source, even! This is a much more efficient way to transmit data. (If you've used BitTorrent, you may have noticed that when you download a file, it fetches it from a bunch of random people; this is the same idea.)
```

> Each block might come from a different source, even!
> f you've used BitTorrent, you may have noticed that when you download a file, it fetches it from a bunch of random people; this is the same idea.)

I'm not sure this was ever implemented on the regular case. Should check if IPFS actually tries to fetch a file from multiple people (i think it will prioritize fetching it from a single source). 

> This is a much more efficient way to transmit data

We should not say this if our implementations cannot back the claim (I think they are slower now).

Reply:
I combined your edits with suggestions from @olizilla . As for whether we can say these things or not....

TODO: _Does IPFS try to fetch files from multiple people_ and _is this a more efficient way to transmit data, ie, do our implementations support this, or is IPFS actually slower when working in this way?_